moz-webgpu-cts/src/shared.rs

```diff
@@ -437,21 +438,28 @@ const SCOPE_DIR_FX_PRIVATE_STR: &str = "testing/web-platform/mozilla";
 const SCOPE_DIR_FX_PRIVATE_COMPONENTS: &[&str] = &["testing", "web-platform", "mozilla"];
 const SCOPE_DIR_FX_PUBLIC_STR: &str = "testing/web-platform";
 const SCOPE_DIR_FX_PUBLIC_COMPONENTS: &[&str] = &["testing", "web-platform"];
+const SCOPE_DIR_SERVO_PUBLIC_STR: &str = "tests/wpt/webgpu";
+const SCOPE_DIR_SERVO_PUBLIC_COMPONENTS: &[&str] = &["tests", "wpt", "webgpu"];
```

Comment:
I have a resolution for this with https://github.com/ErichDonGubler/moz-webgpu-cts/pull/103, which we can save for after this PR.

Reply:
What does private/public mean in this context? Are WebGPU CTS tests meant to be private or public (I considered them as public), but maybe public is meant for upstream WPT tests?

In servo `_webgpu` is alias to `tests/wpt/webgpu/{meta,tests}` which contains WebGPU CTS. An that's it, those are only WebGPU tests we have.

Reply:
@sagudev: Yep, "public" was supposed to mean "upstream WPT". 😅 I definitely wasn't using the right vocabulary; hopefully the table and the PR clarify the details here.

Reply:
Why exactly would upstream path be needed for moz-webgpu-cts? Are there any webgpu test there, or is this some kind of future-proofing for when CTS lands in upstream WPT (or moz-webgpu-cts takes over all expectation updating)?

Reply:
Future-proofing, and also being forced to deal with it early (when I started writing this tool). Due to a fluke in how CI tasks were structured in `mozilla-central`, it was relatively difficult to get reports with only WebGPU tests in WPT test reports. So, I decided to gracefully handle "irrelevant" CTS tests in reports instead.

I don't expect `moz-webgpu-cts` to be useful for general WPT usage. `wpt update` very dynamic in its generation of conditional properties in `wpt update-expectations`, and is configurable at run-time. I don't know who actually uses that amount of power in their daily workflows (CC @jgraham?). However, I suspect that the opinionated, static approach that `moz-webgpu-cts` takes with parsing _and_ properties that get compiled into optimized code (and which allow it to be as fast as it is) may be a significant regression for other users.

Reply:
@sagudev: We could also roll commits from #103 into this PR, if they look good to you.

Reply:
> Due to a fluke in how CI tasks were structured in mozilla-central, it was relatively difficult to get reports with only WebGPU tests in WPT test reports.

In servo we have special CI option to run only subset of wpt tests (only `_webgpu`). 

> So, I decided to gracefully handle "irrelevant" CTS tests in reports instead.

But "public" results/files are not just skipped? They are also read and formatted, IIRC?

Reply:
We now have the right filtering in place for Firefox CI. So, we don't generate reports with unrelated entries anymore.  At the time, though, it was easy to work around it with `git add <scope>/meta/webgpu && git restore .` locally. Even now, we don't fully automate the committing of `expected` properties updated via `moz-webgpu-cts`; there's still a human that looks at things that changed. I don't foresee this changing, as we slowly validate what parts of the CTS are stable for us in Firefox.

> But "public" results/files are not just skipped? They are also read and formatted, IIRC?

Sorry, I didn't expect you to be interested in details, so I've been pretty terse. 😅 You're right; report results for any tests outside `<scope>/meta/webgpu` will still get processed and written out by `moz-webgpu-cts` as with tests it's intended to be used for.

Reply:
> report results for any tests outside <scope>/meta/webgpu will still get processed and written out by moz-webgpu-cts as with tests it's intended to be used for.

That's what's been confusing me the most. In servo there are a lot of "invalid" and not formatted expectations files, so that's why we cannot have `moz-webgpu-cts` try to load them.

Reply:
Is the Servo project actually interested in using `moz-webgpu-cts` in a broader scope? Or is that merely a pain point that's forced you to narrow what metadata `moz-webgpu-cts` is incidentally exposed to? 🤔

Reply:
> Is the Servo project actually interested in using `moz-webgpu-cts` in a broader scope?

Formatting would be useful as I do not think any exists for expectations (or at least we aren't using any), but there is currently nothing planned.

> Or is that merely a pain point that's forced you to narrow what metadata `moz-webgpu-cts` is incidentally exposed to? 🤔

This is exactly what happened.

Reply:
> Formatting would be useful as I do not think any exists for expectations (or at least we aren't using any), but there is currently nothing planned.

I suspect that a general-purpose formatter would be very difficult to get consensus on, as opposed to the very opinionated approach here (CC @jgraham). I don't see any upstream issues in [`wpt`](https://github.com/web-platform-tests/wpt) or [`rfcs`](https://github.com/web-platform-tests/rfcs) for it. However, you _could_ use `whippit` now to create your own parser, and then a custom emitter like [`format_file`](https://github.com/ErichDonGubler/moz-webgpu-cts/blob/6b4d64591b5f54f7f6e4d86c389edf88165e7324/moz-webgpu-cts/src/metadata.rs#L694) to enforce things like property order and spacing. AIUI that's really the full breadth of things you can control, with the metadata language.